IBaseEntity/IBaseEntity.h

```diff
@@ -62,7 +62,7 @@ interface IBaseEntity {
 public:
     friend class CGlobalVars; // allow CGlobalVars to access s_iEntityCount
     IBaseEntity();
-    virtual ~IBaseEntity() { RemoveSelfFromLists(); }

```

Comment:
We no longer manually remove entities and instead do this when the actor is destroyed